Feature/scrum 51 Implemented admin login page

l00188531 opened the pull request:
Implemented the Admin login page, restricting access to view or edit the page without logging in. Created a Flask-Admin group in AWS and added all users to it, granting them access to the application. To log in, users must provide their access key and secret key. All checks passed now
![image](https://github.com/user-attachments/assets/58321df0-4700-4767-9d74-c0f2cdadfad6)

l00188531 committed:
implemented admin login page with admin previliges

l00188531 committed:
implemented admin login page with admin previliges separated GET/POST methods

L00137900 reviewed (changes requested):
Hi Khurram,

Can I ask if there are reasons you have removed the CSRF functionality here? As having CSRF protection which I had added in your previous branch, helps to protect against cross-site request forgery attacks so our application is vulnerable without it.. you had removed swagger endpoints before in a previous PR so I would say you just need to be more careful about what you are removing. If you are unsure about something ask the group, that is what we are here for. Otherwise nice work on this from what I can see here, but CSRF functionality needs re-added.

l00188531 commented:
Hi Ronan,
I had an error related to CSRF while running the application. I'm not sure if any additional dependencies were needed to be installed for this. It might be a good practice to update the requirements.txt file whenever a new dependency is added.
For now, I temporarily removed the CSRF check to proceed, but I will try to restore it. Apologies for removing the swagger endpoint, will review the code and work on restoring it.

L00137900 committed:
Reimplement CSRF functionality and added CSRF to requirements file

L00137900 committed:
Merge pull request #34 from Nausicaa-s-Global-Green-Initiative/Feature/scrum-51

Reimplement CSRF functionality and added CSRF to requirements file

L00137900 reviewed (dismissed):
I switched to your branch to reimplement CSRF functionality and I am seeing no errors, however I have now added it to the requirements.txt file. I am assuming the errors you were seeing is because pip install Flask-WTF needs to be ran in the terminal, I meant to update the slack channel to say to do this like I have done before in previous changes I have made but this time around it just slipped my mind. 

Though just because you are getting errors from something does not mean you automatically remove it, if you are unsure ask. When this goes into dev it will be good to let people know to run pip install boto3 in their terminals. No need to work on restoring swagger endpoints I have already done this in a previous PR: https://github.com/Nausicaa-s-Global-Green-Initiative/Nausica-Project/commit/0f04f8cf42faafba41af1378dcee6f0e7b324367 I am only mentioning it as just to be careful what you remove. All good to approve with restoring CSRF functionality.

l00188531 commented:
Hi Ronan,  
I was working on adding CSRF and Swagger endpoints back to the code, but I noticed in your comments that you’ve already done it. I’ll be more careful next time.  
Thanks for your review!

l00188531 commented:
I  would suggest that add a CSRF token to the `admin_login.html` page as well.  apart from that everything looks good to me.

L00137900 committed:
Add CSRF to Admin_Login as well as fix linter errors

L00137900 committed:
Merge remote-tracking branch 'origin/feature/scrum-51' into feature/scrum-51
Merge remote changes with local

L00137900 committed:
Fix to resolve linter issues in admin_login as well as adding CSRF

- Addressed linter errors by moving template variables to CSS
- Added CSRF token for better security
- Cancel button handling with event listeners for cleaner code

L00137900 commented:
![image](https://github.com/user-attachments/assets/5a05287d-2c91-427f-943e-09d41d3f4ee1)
Yes good suggestion on implementing here as well. I was pretty caught up on getting it added back that it didn't cross my mind so good suggestion. When going to implement CSRF I came across a few errors which I believe to be linter errors but I have resolved these now. 

- I moved background image URL to a CSS variable to avoid linter errors
- Improved cancel button handling with event listeners for cleaner code
- Added CSRF token for enhanced security

L00137900 reviewed (approved):
Looks good with most recent changes

stephenjtyrrell reviewed (commented):
LGTM
